distribution/pull_v2.go

```diff
@@ -858,6 +859,66 @@ func fixManifestLayers(m *schema1.Manifest) error {
 	return nil
 }
 
-func createDownloadFile() (*os.File, error) {
-	return ioutil.TempFile("", "GetImageBlob")
+var v2LayerDataFilename = "v2data"
+
+func resumeOrCreateDownloadFile(progressOutput progress.Output, cachePath string, dgst digest.Digest, id string) (int64, *os.File, digest.Verifier, error) {
```

Comment:
Yeah, probably. I wanted to actually change a bunch of the logic in the Download function, but I didn't fully understand all the error cases, so I didn't end up doing that yet.